src/relay/Web3Response.ts

```diff
@@ -10,18 +10,23 @@ interface BaseWeb3Response<Result> {
   result?: Result
 }
 
-export interface ErrorResponse extends BaseWeb3Response<void> {
+export class ErrorResponse implements BaseWeb3Response<void>, Error {
+  method: Web3Method
   errorCode?: number
   errorMessage: string
+  name: string
+  message: string
+
+  constructor(method: Web3Method, errorMessage: string, errorCode?: number) {
+    this.method = method
+    this.errorMessage = errorMessage
+    this.errorCode = errorCode
+    this.name = method
+    this.message = errorMessage
+  }
 }
 
-export function ErrorResponse(
-  method: Web3Method,
-  errorMessage: string,
-  errorCode?: number
-): ErrorResponse {
-  return { method, errorMessage, errorCode }
-}
+export const USER_REJECTED_REQUEST_ERROR = new Error("User rejected request")
```

Comment:
q - do we need to all caps this?

Reply:
following the pattern on the extension, but open to suggestions.